### ISSUE TITLE
Sweep: [create-lamma] [pnpm] Error: Set OpenAI Key in OPENAI_API_KEY env variable

### ISSUE THREAD

**Issue by RayFernando1337** [run-llama/LlamaIndexTS#182]
### Details

**Steps to reproduce**
1.) Created project using `$ npx create-llama --use-pnpm`
2.) When prompted entered my OpenAI API key directly from OAI
3.) output ran but failed at `pnpm run generate`

**Results**
* It looks like the .env.local file was created with my API key
* Tried to cd into the directory and run `pnpm run generate` but I get the same error
* `npm run generate` also fails with the same error

$ ls -la | grep ".env"
-rw-r--r--   1 ray  staff      67 Nov 14 08:58 .env.local

$ cat .env.local 
OPENAI_API_KEY=sk-BK....

**Invocation**
$ npx create-llama --use-pnpm      
✔ What is your project named? … llama-pdf
✔ Which template would you like to use? › Chat with streaming
✔ Which framework would you like to use? › NextJS
✔ Which UI would you like to use? › Shadcn
✔ Which chat engine would you like to use? › ContextChatEngine
✔ Please provide your OpenAI API key (leave blank to skip): … sk-BK.......
✔ Would you like to use ESLint? … No / Yes
Creating a new LlamaIndex app in /Users/ray/workspace/llama-pdf.

**Output**
```
Using pnpm.

Initializing project with template: streaming 


Using chat engine: context 


Using UI: shadcn 


Installing dependencies:
- ai
- llamaindex
- next
- react
- react-dom
- tailwind-merge
- @radix-ui/react-slot
- class-variance-authority
- lucide-react
- remark
- remark-code-import
- remark-gfm
- remark-math
- react-markdown
- react-syntax-highlighter

Installing devDependencies:
- @types/node
- @types/react
- @types/react-dom
- autoprefixer
- eslint
- eslint-config-next
- postcss
- tailwindcss
- typescript
- @types/react-syntax-highlighter

Downloading registry.npmjs.org/pdf-parse/1.1.1: 11.85 MB/11.85 MB, done
 WARN  1 deprecated subdependencies found: fs-promise@2.0.3
Packages: +592
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
Progress: resolved 600, reused 507, downloaded 85, added 592, done

dependencies:
+ @radix-ui/react-slot 1.0.2
+ ai 2.2.22
+ class-variance-authority 0.7.0
+ llamaindex 0.0.35
+ lucide-react 0.291.0 (0.292.0 is available)
+ next 13.5.6 (14.0.2 is available)
+ react 18.2.0
+ react-dom 18.2.0
+ react-markdown 8.0.7 (9.0.1 is available)
+ react-syntax-highlighter 15.5.0
+ remark 14.0.3 (15.0.1 is available)
+ remark-code-import 1.2.0
+ remark-gfm 3.0.1 (4.0.0 is available)
+ remark-math 5.1.1 (6.0.0 is available)
+ tailwind-merge 2.0.0

devDependencies:
+ @types/node 20.9.0
+ @types/react 18.2.37
+ @types/react-dom 18.2.15
+ @types/react-syntax-highlighter 15.5.10
+ autoprefixer 10.4.16
+ eslint 8.53.0
+ eslint-config-next 13.5.6 (14.0.2 is available)
+ postcss 8.4.31
+ tailwindcss 3.3.5
+ typescript 5.2.2
Done in 22.3s

Created '.env.local' file containing OPENAI_API_KEY

Copying test data to /Users/ray/workspace/llama-pdf/data


Running npm run generate to generate the context data.


> llama-pdf@0.1.0 generate /Users/ray/workspace/llama-pdf
> node app/api/chat/engine/generate.mjs

/Users/ray/workspace/llama-pdf/node_modules/.pnpm/llamaindex@0.0.35/node_modules/llamaindex/dist/index.js:605
      throw new Error("Set OpenAI Key in OPENAI_API_KEY env variable");
            ^

Error: Set OpenAI Key in OPENAI_API_KEY env variable
    at new OpenAISession (/Users/ray/workspace/llama-pdf/node_modules/.pnpm/llamaindex@0.0.35/node_modules/llamaindex/dist/index.js:605:13)
    at getOpenAISession (/Users/ray/workspace/llama-pdf/node_modules/.pnpm/llamaindex@0.0.35/node_modules/llamaindex/dist/index.js:621:15)
    at new OpenAI2 (/Users/ray/workspace/llama-pdf/node_modules/.pnpm/llamaindex@0.0.35/node_modules/llamaindex/dist/index.js:1540:81)
    at serviceContextFromDefaults (/Users/ray/workspace/llama-pdf/node_modules/.pnpm/llamaindex@0.0.35/node_modules/llamaindex/dist/index.js:2083:71)
    at file:///Users/ray/workspace/llama-pdf/app/api/chat/engine/generate.mjs:41:26
    at file:///Users/ray/workspace/llama-pdf/app/api/chat/engine/generate.mjs:48:3
    at ModuleJob.run (node:internal/modules/esm/module_job:194:25)

Node.js v18.18.2
 ELIFECYCLE  Command failed with exit code 1.

Aborting installation.
  pnpm run generate has failed.
```

<details open>
<summary>Checklist</summary>

- [X] Modify `packages/create-llama/templates/index.ts` ✓ https://github.com/run-llama/LlamaIndexTS/commit/bce1694df7e5f069f73f5bc34a5db9c08ab00c2e

![Flowchart](http://24.199.78.105:8082/public/48091abc459b655aaa0950042aa0a21906e70132b964e280e033d276142fc79d_182_flowchart.svg)
</details>

**Comment by yisding** [comment]
Thanks for the bug report! Fixing now.

**Comment by yisding** [comment]
OK should be fixed in 0.0.5

**Comment by emapeire** [comment]
I have the same issue with the latest update, and still does not work. When I ran: `npm run generate`, I got this message:

```
npm run generate

> eloquent-js-chat@0.1.0 generate
> node app/api/chat/engine/generate.mjs

/Users/emapeire/Workspace/AI/eloquent-js-chat/node_modules/llamaindex/dist/index.js:605
      throw new Error("Set OpenAI Key in OPENAI_API_KEY env variable");
            ^

Error: Set OpenAI Key in OPENAI_API_KEY env variable
    at new OpenAISession (/Users/emapeire/Workspace/AI/eloquent-js-chat/node_modules/llamaindex/dist/index.js:605:13)
    at getOpenAISession (/Users/emapeire/Workspace/AI/eloquent-js-chat/node_modules/llamaindex/dist/index.js:621:15)
    at new OpenAI2 (/Users/emapeire/Workspace/AI/eloquent-js-chat/node_modules/llamaindex/dist/index.js:1540:81)
    at serviceContextFromDefaults (/Users/emapeire/Workspace/AI/eloquent-js-chat/node_modules/llamaindex/dist/index.js:2083:71)
    at file:///Users/emapeire/Workspace/AI/eloquent-js-chat/app/api/chat/engine/generate.mjs:41:26
    at file:///Users/emapeire/Workspace/AI/eloquent-js-chat/app/api/chat/engine/generate.mjs:48:3
    at ModuleJob.run (node:internal/modules/esm/module_job:217:25)
    at async ModuleLoader.import (node:internal/modules/esm/loader:316:24)
    at async loadESM (node:internal/process/esm_loader:34:7)
    at async handleMainPromise (node:internal/modules/run_main:66:12)

Node.js v20.9.0
```

However, I do not why happen this, because I have the `.env.local` file with my `OPEN_API_KEY` key.
